drivers/serial/uart_nrfx_uart.c

```diff
@@ -38,6 +38,33 @@ static inline const struct uart_nrfx_config *get_dev_config(struct device *dev)
 	return dev->config->config_info;
 }
 
+#ifdef CONFIG_UART_ASYNC_API
+static struct {
+	uart_callback_t callback;
+	void *user_data;
+
+	bool rx_enabled;
```

Comment:
I understand the desire to have rx/tx params grouped, but IMHO worth putting them together to have better struct packing.

Reply:
Just moved bool values close to each other, all other values are 4 byte aligned.